processor/spanmetricsprocessor/processor.go

```diff
@@ -104,6 +108,17 @@ func newProcessor(logger *zap.Logger, config config.Processor, nextConsumer cons
 		return nil, err
 	}
 
+	if pConfig.MetricKeyToDimensionsCacheSize <= 0 {
+		return nil, fmt.Errorf(
+			"invalid cache size: %v, the maximum number of the items in the cache should be positive",
+			pConfig.MetricKeyToDimensionsCacheSize,
+		)
+	}
```

Comment:
This is code block necessary? Based on [this test](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6503/files#diff-d6d50caba43f477107ca6d4f0409f2400586cac0104bd697e61383113d30b9aaR41) it looks like `lru.NewWithEvict` will return an error if `size <= 0`: [reference link](https://github.com/hashicorp/golang-lru/blob/80c98217689d6df152309d574ccc682b21dc802c/simplelru/lru.go#L27-L29).

Reply:
I think we should keep it. The error from `lru.NewWithEvic` would be a lower-level error at the cache implementation layer. The error from here is about the processor configuration validation.

Reply:
Okay, I agree, the error message here is more meaningful/useful. 👍🏼